src/effects/nyquist/Nyquist.cpp

```diff
@@ -195,6 +195,8 @@ NyquistEffect::NyquistEffect(const wxString &fName)
 
    if (!mOK && mInitError.empty())
       mInitError = XO("Ill-formed Nyquist plug-in header");
+
+   mHelpFileExists = CheckHelpPage().first;
```

Comment:
What other member variables does CheckHelpPage depend on?  Answer: mHelpFile

Where is mHelpFile reassigned?  Answer: NyquistEffect::Parse and NyquistEffect::ParseProgram

Are these functions called only in the constructor of NyquistEffect?  Answer: no (explore the call graph)

Therefore, the end of the constructor is not the correct place to assign the variable.

But, the end of ParseProgram would be the correct place.  Because Parse is called only inside ParseProgram.  So that whenever mHelpFile might change, the update of mHelpFileExists will be correct.

Reply:
Good catch - I was making the wrong assumption that `ParseFile `(which calls `ParseProgram`) was called in the constructor only, but it is called in ::`Init `too.

Reply:
done